File: PodFul.FileDelivery.IntegrationTests2/FileDeliveryPoint_IntegrationTests.fs
namespace PodFul.FileDelivery.IntegrationTests

open System
open System.IO
open Jabberwocky.Toolkit.IO
open NUnit.Framework
open PodFul.FileDelivery

type FileDeliveryPoint_IntegrationTests() =

    let mutable workingDirectory = String.Empty;

    [<SetUp>]
    member public this.SetupBeforeEachTest() =
        workingDirectory <- Path.Combine(Path.GetTempPath(), "FileDeliveryPoint_IntegrationTests")
        DirectoryOperations.EnsureDirectoryIsEmpty(workingDirectory) 

    [<Test>]
    member public this.Test() =
        let filePath = Path.Combine(workingDirectory, Path.GetTempFileName())
        let destinationPath = Path.Combine(workingDirectory, "Destination")
        let fileDeliveryPoint = new FileDeliveryPoint(destinationPath, )

        fileDeliveryPoint.Deliver(filePath, "file")

File: PodFul.FileDelivery.IntegrationTests/FileDeliveryPoint_IntegrationTests.cs
﻿
namespace PodFul.FileDelivery.IntegrationTests
{
  using System;
  using System.IO;
  using FileDelivery;
  using Jabberwocky.Toolkit.IO;
  using NUnit.Framework;
  using Shouldly;

  [TestFixture]
  public class FileDeliveryPoint_IntegrationTests
  {
    private String workingDirectory;

    #region Methods
    [SetUp]
    public void SetupBeforeEachTest()
    {
      this.workingDirectory = Path.Combine(Path.GetTempPath(), "FileDeliveryPoint_IntegrationTests");
      DirectoryOperations.EnsureDirectoryIsEmpty(this.workingDirectory);
    }

    [Test]
    public void Deliver_DeliveringFileToDirectory_FileExists()
    {
      // Arrange
      var fileName = Path.GetRandomFileName();
      var filePath = Path.Combine(workingDirectory, fileName);
      var fileStream = File.Create(filePath);
      fileStream.WriteByte(1);
      fileStream.Close();

      var destinationPath = Path.Combine(workingDirectory);
      var expectedDestinationFilePath = Path.Combine(destinationPath, DateTime.Now.ToString("dd-MM-yyyy") + "_1");
      expectedDestinationFilePath = Path.Combine(expectedDestinationFilePath, fileName);

      var fileDeliveryPoint = new FileDeliveryPoint(destinationPath, null, null);

      // Act
      fileDeliveryPoint.Deliver(filePath, "file");

      // Assert
      File.Exists(expectedDestinationFilePath).ShouldBeTrue();
    }
    #endregion 
  }
}
